src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1166,6 +1166,58 @@ internal virtual void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleB
         /// </summary>
         public abstract bool AreLocalsZeroed { get; }
 
+        internal virtual bool IsNullableEnabled() => false;
+
+        /// <summary>
+        /// If this method is a constructor, includes the nullable context for any
+        /// fields from the containing type that are set in this constructor.
+        /// </summary>
+        internal bool IsNullableEnabledCore(bool? isEnabledInBody)
+        {
+            if (isEnabledInBody == true)
+            {
+                return true;
+            }
+
+            bool? result = isEnabledInBody;
+            if (this.IncludeFieldInitializersInBody() &&
+                ContainingType is SourceMemberContainerTypeSymbol containingType)
+            {
+                foreach (var member in containingType.GetMembersUnordered())
```

Comment:
How do we protect against the case where a constructor is added, `IsNullableEnabledCore` is called and then another member is added to `containingType`? That worries me a bit.

Reply:
`GetMembersUnordered()` should be immutable.

---
In reply to: [545951290](https://github.com/dotnet/roslyn/pull/49953#discussion_r545951290) [](ancestors = 545951290)